repos/system_upgrade/el7toel8/actors/initramdiskgenerator/libraries/initramgen.py

```diff
@@ -0,0 +1,97 @@
+import os
+import shutil
+
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.common import dnfplugin, mounting
+from leapp.libraries.stdlib import api
+from leapp.models import BootContent, RequiredUpgradeInitramPackages, TargetUserSpaceInfo, UpgradeDracutModule
+
+INITRAM_GEN_SCRIPT_NAME = 'generate-initram.sh'
+
+
+def _reinstall_leapp_repository_hint():
+    """
+    Convenience function for creating a detail for StopActorExecutionError with a hint to reinstall the
+    leapp-repository package
+    """
+    return {
+        'hint': 'Try to reinstall the `leapp-repository` package.'
+    }
+
+
+def copy_dracut_modules(context, modules):
+    """
+    Copies our dracut modules into the target userspace.
+    """
+    try:
+        shutil.rmtree(context.full_path('/dracut'))
+    except EnvironmentError:
+        pass
+    for module in modules:
+        try:
+            context.copytree_to(module.module_path, os.path.join('/dracut', os.path.basename(module.module_path)))
+        except shutil.Error as e:
+            api.current_logger().error('Failed to copy dracut module "{name}" from "{source}" to "{target}"'.format(
+                name=module.name, source=module.module_path, target=context.full_path('/dracut')), exc_info=True)
+            raise StopActorExecutionError(
+                message='Failed to install dracut modules required in the initram. Error: {}'.format(e.message)
+            )
+
+
+def _get_dracut_modules():
+    return list(api.consume(UpgradeDracutModule))
```

Comment:
this is wrong. You cannot cosume stuff like that in shared libraries - at least, you cannot do that without checks that the actor is consuming this type of message. Otherwise someone would use the library and nothing will happen because the list of consumed messages will be empty. This is problem on several other places (all consumes in shared libraries)

Reply:
This particular one is actually in a private library.

Reply:
we should even consider putting some guidance about that into the best practices or somewhere

Reply:
@vinzenz ah. yes, I marked the wrong one. Sorry, but the problem is in two shared libraries.